File: src/optic/lua/ssa/instructions/GetVarargs.java
package optic.lua.ssa.instructions;

import optic.lua.ssa.*;

public final class GetVarargs implements Step {
	private final Register to;

	public GetVarargs(Register to) {
		this.to = to;
	}

	@Override
	public String toString() {
		return "varargs " + to + " = ...";
	}
}

File: src/optic/lua/asm/instructions/GetVarargs.kt
package optic.lua.asm.instructions

import optic.lua.asm.Register
import optic.lua.asm.Step
import optic.lua.asm.StepVisitor

class GetVarargs(val to: Register) : Step {
    override fun <T : Any, X : Throwable> accept(visitor: StepVisitor<T, X>): T = visitor.visitGetVarargs(to)

    override fun toString(): String {
        return "varargs $to = ..."
    }

    override fun modified(): Register? = to
}
